### PR TITLE
[WiP] Migrate Paste website from Gatsby to NextJS

### PR DESCRIPTION
**Pull request by TheSisb**
We've decided to migrate our documentation website from Gatsby to NextJS. This decision was made due to several factors, but the two primary concerns were the slow build times and the fragility of the builds.

### Done
- [x] Create a paste-docs folder with a TS+NextJS codebase 
- [x] Update repo's package.json file with new start and build commands.
- [x] Upgrade React from 16.8 to 16.14 to leverage new JSX transforms  (https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform)
- [x] Fix all missing peerDeps warnings in Paste
- [x] Upgrade NextJS in monorepo from 10.0 to 10.2
- [x] Port our homepage
- [x] Port out the SiteWrapper code 

### Todo
- [ ] Fix google analytics tracking code (search code for TODO)
- [ ] Fix docsearch on SSR in NextJS (look into how we solved this in Gatsby)
- [ ] Build site sidebar
- [ ] Build all the doc pages
- [ ] Build roadmap page + airtable integration
- [ ] Fix SVG imports
- [ ] Replace all gatsby Links with nextJS Links
- [ ] Replace all gatsby images with NextJS images
- [ ] ???

### Polish at the end

- [ ] Remove the `paste-website` folder and rename `paste-docs` to `paste-website`
- [ ] Update main `package.json` scripts to refer to paste-website only, removing paste-docs mentions
- [ ] Make this all work on CI + CD
- [ ] ???

### CODE REVIEW

**Comment by TheSisb** [comment]
tl;dr: Halting this work until NextJS 10, Emotion 11, MDX, and Typescript 4 mature some more and allow better interplay.

Long explaination:

Getting NextJS v10.x up and running revealed that it used React v17, which has a [new JSX Transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) that didn't mesh well in our monorepo.  This is because our monorepo uses React v16.8 which doesn't have this transform, and all monorepo packages try re-using the same version of React.  Fortunately, the react team released v16.14 which does have this new transform without needing to perform a major upgrade, and so I made that minor upgrade which let NextJS run properly.

In order to get emotion working well in NextJS I needed to create a babelrc file with the new recommendations for emotion in NextJS:  https://emotion.sh/docs/css-prop#babel-preset (cmd + f "next"). This worked well at first, but broke when I tried adding MDX to the equation.

I initially tried using [next-mdx-remote](https://github.com/hashicorp/next-mdx-remote) but it [doesn't currently work in monorepos](https://github.com/hashicorp/next-mdx-remote/pull/135).  I tried `next-mdx-enhanced` and just plain `@mdx-js/react`, but neither worked due to a mixture of Emotion and this new JSX transform. The exact error is:

```shell
importSource cannot be set when runtime is classic.
> 1 |
    | ^
  2 | import React from 'react'
  3 | import { mdx } from '@mdx-js/react'
  4 |
    at transformFile.next (<anonymous>)
    at run.next (<anonymous>)
``` 
Emotion needs to set the runtime to work in NextJS, but MDX is all kinds of confused for some unknown reason.  

I played around with upgrading emotion but Typescript started to complain.  Anytime we're upgrading too many core libraries at once, problems arise.  We should probably update React and Emotion before trying to use NextJS again.

**Comment by TheSisb** [comment]
Closing until we're ready to re-tackle this.
